beacon-chain/sync/validate_aggregate_proof.go

```diff
@@ -61,8 +59,25 @@ func (r *Service) validateAggregateAndProof(ctx context.Context, pid peer.ID, ms
 		return false
 	}
 
-	// Verify the block being voted for passes validation. The block should have passed validation if it's in the DB.
-	if !r.db.HasBlock(ctx, bytesutil.ToBytes32(m.Aggregate.Data.BeaconBlockRoot)) {
+	if !r.validateAggregatedAtt(ctx, m) {
+		return false
+	}
+
+	msg.ValidatorData = m
+
+	return true
+}
+
+func (r *Service) validateAggregatedAtt(ctx context.Context, a *ethpb.AggregateAttestationAndProof) bool {
```

Comment:
Why refactor this into another method? `validateAggregateAndProof` describes the behavior of `validateAggregatedAtt` or are they different?

Reply:
I want to re-use second half of `validateAggregateAndProof` to validate pending attestations when the missing block arrives. And I dont really want to use `validateAggregateAndProof` to validate pending attestation, it takes in `pid peer.ID, msg *pubsub.Message` which makes it hard to satisfy (pending attestation can also be unaggregated)


With the new refactor:

```go
func validateAggregateAndProof() {

check if it is syncing

decodePubsubMessage

check if the attestation is already in the pool

validateAggregatedAtt // This is reused for processing pending attestation

}
```

Reply:
Sounds good thanks